### PR TITLE
UML-1282 - Lookup accounts by LPA ID

### PR DESCRIPTION
**Pull request by andrewpearce-digital**
# Purpose

- [x] A search by Email Address returns a list of LPA numbers that are added to the account.
- [x] A search by LPA number that returns a list of email addresses that it is assigned too.

Fixes UML-1282

## Approach

- [x] Refactor existing script to make space for new function
- [x] Add arguments to search by LPA ID
- [x] Paginate a scan operation to iterate over LPA IDs
- [x] Refactor to remove duplication 
- [x] Refactor to optimise DynamoDB operations
- [x] Produce JSON output

## Learning

_Any tips and tricks, blog posts or tools which helped you. Plus anything notable you've discovered about the Use a Lasting Power of Attorney service_

## Checklist

* [x] I have performed a self-review of my own code
* [x] I have added relevant logging with appropriate levels to my code
* [x] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [ ] I have added tests to prove my work
* [ ] I have added welsh translation tags and updated translation files
* [ ] I have run an accessibility tool on any pages I have made changes to and fixed any issues found
* [ ] The product team have tested these changes

### CODE REVIEW

**Review by johnnolan — DISMISSED** [review]
Tested and used locally, both lpaid and email both work

File: scripts/analysis_scripts/lookup_account.py [hunk 1]
```diff
@@ -124,17 +140,21 @@ def main():
                         default="",
                         help="Email address to look up")
 
+    arguments.add_argument("--lpa_id",
+                        default="",
+                        help="Email address to look up")
```
Comment (cooperaj) [inline scripts/analysis_scripts/lookup_account.py:145]: help text not right?

Comment (andrewpearce-digital) [reply scripts/analysis_scripts/lookup_account.py:145]: Thanks, copy-paste error there. I've got a ton more refactoring to do today, so will sort that in the next commit

**Comment by andrewpearce-digital** [comment]
So right now, this script will return separate blocks of json (printed as plain text) for each account match found.
This seems fine for account lookups by email address as we limit accounts to one per email address.
Lookups by LPA ID are likely to return multiple accounts. So I'll look at how to produce a single structure for that next

File: scripts/analysis_scripts/lookup_account.py [hunk 2]
```diff
@@ -3,12 +3,14 @@
 from datetime import date
```
Comment (johnnolan) [inline scripts/analysis_scripts/lookup_account.py:3]: I believe this can now be removed as its no longer required

**Review by johnnolan — APPROVED** [review]

**Pull request merged**